src/aio/aio_theme_switch.py

```diff
@@ -61,118 +59,60 @@ def __init__(
         - ThemeSwitchAIO.ids.store(aio_id)
         """
 
-        from dash_bootstrap_templates import load_figure_template
-
-        load_figure_template(dbc_themes_lowercase)
-
-        if aio_id is None:
-            aio_id = str(uuid.uuid4())
-
-        if themes is None:
-            themes = [dbc.themes.BOOTSTRAP, dbc.themes.CYBORG]
+        # init icons and switch_props
         if icons is None:
             icons = {"left": "fa fa-moon", "right": "fa fa-sun"}
-
-        switch_props = switch_props.copy()
-        if "value" not in switch_props:
-            switch_props["value"] = True
-        if "className" not in switch_props:
-            switch_props["className"] = "d-inline-block ms-1"
+        if switch_props is None:
+            switch_props = {}
+        # set "value" and "className" if they don't exist
+        switch_props.setdefault("value", True)
+        switch_props.setdefault("className", "d-inline-block ms-1")
+
+        # add fontawesome resource for the icons and dbc.css to apply Bootstrap theme to dcc, DataTable and Dash AG Grid
+        app = get_app()
+        app.config.external_stylesheets += [
+            "https://use.fontawesome.com/releases/v5.15.4/css/all.css",
+            # "https://cdn.jsdelivr.net/gh/AnnMarieW/dash-bootstrap-templates/dbc.min.css"
+        ]
+
+        # if using custom themes in /assets, filter them out to not be automatically imported by Dash
+        # and let the switch handle them.
+        # append "|" if assets_ignore has already regex rules
+        if app.config.assets_ignore:
+            app.config.assets_ignore += "|"
+        # add the themes to ignore in /assets
+        if isinstance(themes, str):
+            app.config.assets_ignore += f'{themes.replace("/assets/", "")}'
+        else:
+            app.config.assets_ignore += f'{themes[0].replace("/assets/", "")}|{themes[1].replace("/assets/", "")}'
+
+        # make all dash_bootstrap_templates templates available to plotly figures
+        load_figure_template('all')
```

Comment:
Good refactor!  

Note to self - is it really necessary to load all the figure templates?  Is it too much overhead give that there are now 52 of them and this component only switches between 2 themes?  However, it might be a breaking change to only load those associated with the themes.

Reply:
If we don't load all the templates in the AIO, we can load only the 2 used in the app:
```python
from dash_bootstrap_templates import read_template

# Figure templates from dash_bootstrap_templates
fig_templates = ("minty", "cyborg")
# Make templates available to plotly figures
for template in fig_templates:
    read_template(template)


# Switch figure themes
@callback(
    Output("theme_switch-graph", "figure"),
    Input(ThemeSwitchAIO.ids.switch("theme"), "value"),
)
def update_figure_template(switch_on):
    return px.bar(
        df, x="Fruit", y="Amount", color="City", barmode="group",
        template=fig_templates[0] if switch_on else fig_templates[1]
    )
```

Reply:
But indeed that means a little breaking change as we have to use `read_template(template)` 🤔 
What do you prefer?

Reply:
I've been pondering that question - I think it's best to leave it as-is for now and continue to load all the templates.  I'll open an issue to start collecting things that could be included in V2 and add this to the list :-)

Reply:
So I think I'm good to go with the modifications, I'm going to push the commit ! 🚀